akka-http/src/main/scala/akka/http/scaladsl/marshalling/PredefinedToEntityMarshallers.scala

```diff
@@ -43,7 +43,18 @@ trait PredefinedToEntityMarshallers extends MultipartMarshallers {
 
   implicit val StringMarshaller: ToEntityMarshaller[String] = stringMarshaller(`text/plain`)
   def stringMarshaller(mediaType: MediaType.WithOpenCharset): ToEntityMarshaller[String] =
-    Marshaller.withOpenCharset(mediaType) { (s, cs) => HttpEntity(mediaType withCharset cs, s) }
+    Marshaller.withOpenCharset(mediaType) { (s, cs) =>
+      val charset = try {
+        cs.nioCharset // throws if value is invalid
+        cs
+      } catch {
+        case _: java.nio.charset.UnsupportedCharsetException =>
+          // request contained an invalid charset name, fall back to UTF-8
+          HttpCharsets.`UTF-8`
```

Comment:
ok, so this would be the same as if the header was not specified? effectively ignoring the header if it's invalid, which is along the lines of the recommendation in https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Charset